crates/interpreter/src/gas.rs

```diff
@@ -94,8 +94,12 @@ impl Gas {
     ///
     /// Related to EIP-3529: Reduction in refunds
     #[inline]
-    pub fn set_final_refund<SPEC: Spec>(&mut self) {
-        let max_refund_quotient = if SPEC::enabled(LONDON) { 5 } else { 2 };
+    pub fn set_final_refund(&mut self, spec: SpecId) {
```

Comment:
Would put here just a bool flag, `is_london` here

Reply:
Don't think that makes much sense if all callers will end up doing `spec.is_enabled_in(SpecId::LONDON)`

Reply:
It is more code for upper layers, but it is easier to see that this function depends only on `SpecId::LONDON`. This is the only function in `Gas` that is hardfork related.